### PR TITLE
Add set, and histogram reporting to aerospike telegraf plugin

### PR DESCRIPTION
**Pull request by grossjo**
refactored to allow non integration tests
fix tests to work with aerospike 4.9.0.11

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by ssoroka** [comment]
Thanks! 

Looks like github.com/influxdata/telegraf/plugins/inputs/net_response failed, but only on darwin. Going to rerun it to see if it was just an intermittent fluke.

Will review

**Comment by grossjo** [comment]
I attempted to make it backwards compatible, where all new features are opt-in.
Which should be good for any existing users of this plugin.

**Comment by grossjo** [comment]
Hi,
Just wondering what it would take to get this merged in?

File: plugins/inputs/aerospike/aerospike.go [hunk 1]
```diff
@@ -68,6 +102,14 @@ func (a *Aerospike) Gather(acc telegraf.Accumulator) error {
 		a.initialized = true
 	}
 
+	if a.NumberHistogramBuckets == 0 {
+		a.NumberHistogramBuckets = 10
+	} else if a.NumberHistogramBuckets > 100 {
+		a.NumberHistogramBuckets = 100
+	} else if a.NumberHistogramBuckets < 1 {
+		a.NumberHistogramBuckets = 10
+	}
```
Comment (ssoroka) [inline plugins/inputs/aerospike/aerospike.go:111]: You inherited this, but there's an opportunity to clean it up. plugins can have an Init() function that allows you to do init and config related things that runs once and before the first Gather call.

Just add `func (a *Aerospike) Init() error {` and you can move all this setup-related stuff there.

File: plugins/inputs/aerospike/aerospike.go [hunk 2]
```diff
@@ -108,51 +150,323 @@ func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) erro
...
-				if len(parts) < 2 {
-					continue
+
+			if pieces[0] == "buckets" {
+				buckets := strings.Split(pieces[1], ",")
+
+				// Normalize incase of less buckets than expected
+				numRecordsPerBucket := 1
+				if len(buckets) > a.NumberHistogramBuckets {
+					numRecordsPerBucket = int(math.Ceil((float64(len(buckets)) / float64(a.NumberHistogramBuckets))))
```
Comment (ssoroka) [inline plugins/inputs/aerospike/aerospike.go:430]: might end up with the last bucket being slightly unbalanced if you pick a divisor that's not a round multiple, eg 7.. Probably not a big deal though. Might be worth mentioning in the docs that NumberHistogramBuckets should divide evenly into 100 or len(buckets) to have a balanced histogram?

**Comment by grossjo** [comment]
I'll add the init function,  Didn't know about that.
Give me a day.

File: plugins/inputs/aerospike/aerospike.go (hunk 1 above)
Comment (grossjo) [reply plugins/inputs/aerospike/aerospike.go:111]: maybe next time we do more work, we'll update this. Did not have time this iteration

**Comment by grossjo** [comment]
All changes we are going to do this time are pushed up.  If you are good with this PR, feel free to merge.

**Comment by ssoroka** [comment]
Thanks, Josh :)

**Pull request merged**